hearth/lib/hearth/dns/host_resolver.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Hearth
+  module DNS
+    # Resolve a host name to an address.
+    # @api private
+    class HostResolver
+      def resolve_address(options = {})
+        addrinfo_list = addrinfo(options)
+        ipv6 = ipv6_addr(addrinfo_list, options) if use_ipv6?
+        ipv4 = ipv4_addr(addrinfo_list, options)
+        [ipv6, ipv4]
+      end
+
+      private
+
+      def addrinfo(options)
+        Addrinfo.getaddrinfo(
+          options[:nodename],
+          options[:service],
+          options[:family],
+          options[:socktype] || :SOCK_STREAM,
+          options[:protocol],
+          options[:flags]
+        )
+      end
+
+      def ipv4_addr(addrinfo_list, options)
+        addr = addrinfo_list.find(&:ipv4?)
```

Comment:
this is always going to find the first address right?  We want to return a different address each time this is called I thought?

Reply:
getaddrinfo will return new sets of addresses each time. Taking the first is recommended.